plugins/resources/unixfilesystem/src/unixfilesystem.cpp

```diff
@@ -83,7 +86,91 @@
 const std::string DEFAULT_VAULT_DIR_MODE( "default_vault_directory_mode_kw" );
 const std::string HIGH_WATER_MARK( "high_water_mark" ); // no longer used
 const std::string REQUIRED_FREE_INODES_FOR_CREATE("required_free_inodes_for_create"); // no longer used
+const std::string HOST_MODE("host_mode");
 
+namespace
+{
+    auto is_operating_in_detached_mode(irods::plugin_property_map& prop_map) -> bool
+    {
+        std::string host_mode_str;
+
+        irods::error ret = prop_map.get<std::string>(HOST_MODE, host_mode_str);
+        if (ret.ok()) {
+            return boost::iequals(host_mode_str.c_str(), "detached");
+        }
+
+        // default is attached mode
+        return false;
+    }
+} // namespace
+
+// The return value is always SUCCESS() because the errors encountered are to
+// be treated as warnings and no fatal errors.
+auto unix_file_start_operation(irods::plugin_property_map& prop_map) -> irods::error
+{
+    namespace logger = irods::experimental::log;
+
+    if (!is_operating_in_detached_mode(prop_map)) {
+        // Detached mode is disabled.  Return early.
+        return SUCCESS();
+    }
+
+    // check if we are in the host list
+    char local_hostname[MAX_NAME_LEN];
+    gethostname(local_hostname, MAX_NAME_LEN);
+
+    auto resource_hostname_optional = resolve_hostname(local_hostname, hostname_resolution_scheme::match_preferred);
+
+    std::string resource_hostname =
+        resource_hostname_optional ? std::move(*resource_hostname_optional) : local_hostname;
+
```

Comment:
Remove line.

Reply:
If you want I can remove every empty line in the entire file.

Reply:
oh dear.

Reply:
done